src/decoder/unfiltering_buffer.rs

```diff
@@ -26,6 +26,9 @@ pub(crate) struct UnfilteringBuffer {
     available: usize,
     /// The number of bytes before we shift the buffer back.
     shift_back_limit: usize,
+    /// Size of the previously unfiltered row. For interlaced images this changes over time, from
+    /// pass to pass. Also if we get other subframes.
+    unfiltered_len: usize,
```

Comment:
Would it make sense to call this `rowlen`?

Or maybe this field can be removed, and instead we can add `rowlen` parameter to `fn prev_row`?  Would that work?

PS. I assume that this field wasn't needed in the past, because `curr_start - prev_start` was sufficient before the changes in this PR?  Can you please confirm?

Reply:
We used `rowlen` to refer to the filter byte plus the byte data, whereas here we store `rowlen - 1`. I think it would be confusing to. But I take away that `unfiltered_len` is not the best name either.

In the one call where we handle `prev_row` there was an additional indexing by `[..rowlen - 1]`. I'm not completely sure if the necessity may have changed over time as well. So hard to confirm completely. It is definitely needed after this PR (and cleaner imo).

Reply:
Thanks for explaining.  I actually missed/forgot that `rowlen` refers to something slightly different - I agree that using `rowlen` name here would have been confusing.

So, maybe let's just tweak the comment?:

```
/// Size of an unfiltered row (without the `FilterType` byte / `rowlen - 1`).
/// This may vary over time depending on the size of the frame being
/// currently decoded, and on the interlacing pass.
```